### PR TITLE
Separate included files for Qt4 and Qt5 libraries

### LINKED ISSUE #23667: Slim down the MantidQtCommonWidgetsQt5 library
Parent issue: #23263

Try and slim down the MantidQtCommonWidgetsQt5 library. Most of the files compiled in the Qt5 version of the library are not used. Slim down the Qt5 build to just those classes that are used.

### PR DESCRIPTION
**Pull request by dtasev**
**Description of work.**

Separates the current function call for building the Qt 4 and 5 libraries into two
separate function calls - for Qt4 and Qt5. The Qt5 call only includes
files that are necessary for successful compilation.

This reduced the size of the `MantidQtWidgetsCommonQt5.dll` file by 44.9%

| Name  | MantidQtWidgetsCommonQt4.dll | MantidQtWidgetsCommonQt5.dll |
| ------------- | ------------- | ------------- |
| Original | 5009.5KB  | 5009.5KB |
| Trimmed Qt5 | 5009.5KB  | 2827.5KB |
| Without PeriodicTableWidget | 4855.5KB  | 2673.5KB |

- `PeriodicTableWidget` has been moved to `qt\widgets\plugins\algorithm_dialogs\CMakeLists.txt`
- Tabs have been removed from `qt\widgets\common\CMakeLists.txt` and replaced with 2 spaces.

**Report to:** @DTasev 

**To test:**
Check if Mantid compiles and runs successfully.

Fixes #23667 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/widgets/common/CMakeLists.txt [hunk 1]
```diff
@@ -94,74 +269,75 @@ set ( SRC_FILES
...
+  src/MantidTreeModel.cpp
+  src/MantidTreeWidget.cpp
+  src/MantidTreeWidgetItem.cpp
+  src/MantidWSIndexDialog.cpp
+  src/MessageDisplay.cpp
+  src/MultifitSetupDialog.cpp
+  src/MuonFitDataSelector.cpp
+  src/MuonFitPropertyBrowser.cpp
+  src/MuonFunctionBrowser.cpp
+  # todo check if it can be removed from qt4
```
Comment (martyngigg) [inline qt/widgets/common/CMakeLists.txt]: Can we remove the todo comments, please?

File: qt/widgets/common/CMakeLists.txt [hunk 2]
```diff
@@ -231,7 +407,8 @@ set ( MOC_FILES
   inc/MantidQtWidgets/Common/MuonFitPropertyBrowser.h
   inc/MantidQtWidgets/Common/MuonFunctionBrowser.h
   inc/MantidQtWidgets/Common/MWDiag.h
-  inc/MantidQtWidgets/Common/PeriodicTableWidget.h
+  # todo check if it can be removed from qt4
+
```
Comment (martyngigg) [inline qt/widgets/common/CMakeLists.txt]: Same comment as above.

File: qt/widgets/common/CMakeLists.txt [hunk 3]
```diff
@@ -441,7 +617,8 @@ set ( UI_FILES
   inc/MantidQtWidgets/Common/SlicingAlgorithmDialog.ui
   inc/MantidQtWidgets/Common/SlitCalculator.ui
   inc/MantidQtWidgets/Common/UserFunctionDialog.ui
-  inc/MantidQtWidgets/Common/PeriodicTableWidget.ui
+  # todo check if it can be removed from qt4
```
Comment (martyngigg) [inline qt/widgets/common/CMakeLists.txt]: And again...

File: qt/widgets/plugins/algorithm_dialogs/inc/MantidQtWidgets/Plugins/AlgorithmDialogs/PeriodicTableWidget.h [hunk 4]
```diff
@@ -32,7 +32,7 @@
   Code Documentation is available at: <http://doxygen.mantidproject.org>
 */
 
-class EXPORT_OPT_MANTIDQT_COMMON PeriodicTableWidget : public QWidget {
```
Comment (martyngigg) [inline qt/widgets/plugins/algorithm_dialogs/inc/MantidQtWidgets/Plugins/AlgorithmDialogs/PeriodicTableWidget.h]: :+1:

**Review by martyngigg — CHANGES_REQUESTED** [review]
This looks good. It will allow us to better track what is actually required for the Qt5 build and when we come to remove the Qt4 code we will not accidentally sweep up lots of old code.

Just a couple of minor changes requested to remove some outdated comments.

File: qt/widgets/common/CMakeLists.txt (hunk 1 above)
Comment (DTasev) [reply qt/widgets/common/CMakeLists.txt]: Comments have been removed 👍

**Review by martyngigg — APPROVED** [review]

**Pull request merged**